lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1702,6 +1702,7 @@ public function get_collection_params() {
 		$params['filter'] = array(
 			'description'       => __( 'Use WP Query arguments to modify the response; private query vars require appropriate authorization.' ),
 		);
+		$params['filter[posts_per_page]'] = $params['per_page'];
```

Comment:
Okay, now I see what you are doing.  I am not sure whether this is the best approach to solving this issue.  It is basically like adding a new top level parameter but as an array which is clever, but somewhat confusing, to me at least.  This is why the lines needed to be added in the tests for query params.  This approach definitely works and adds minimal lines of code but just feels just a bit odd to me; definitely clever though! :smile: I think separately validating the filter for various use cases, would be better than using rest_validate_request_arg but that is just my opinion.